### ISSUE TITLE
Make plugin page viewable to users with no sites

### ISSUE THREAD

**Issue by autumnfjeld** [Automattic/wp-calypso#82549]
# Description
When a user with no sites views a plugin logged in, they get a view like this. We need to correct this so that the plugin listing shows for any user, even if they do not have any sites.

![image](https://github.com/Automattic/wp-calypso/assets/5665959/8ec3e467-89da-4b2a-911e-dfabcc252ee1)

https://www.loom.com/share/f71deadefce943628128c760a6d0d8db?sid=cd2492cb-8ae7-4c51-a2df-4f7db119b9d3

This is a high priority request. I believe it was reported on social media, discussed in public and Matt would like a fix to be prioritized. 

# Discussions
* p1696317418348179-slack-C02JPCHEKDY
* p1696342802922279-slack-C02JPCHEKDY
* Context: p1696342931929139-slack-C02JPCHEKDY

# Knowns
Investigation  so far by @Copons 
From p1696345211454279-slack-C02JPCHEKDY

The NoSitesMessage screen is part of the Calypso routing itself, specifically of the siteSelection step.
That step is an essential part of almost all Calypso routes (AFAICS except Reader and Domains).
Its purpose is to check the URL, and either select the current site or send to the site selection screen. Or show the “no sites” screen.
All the routes using it are built with the idea that the user will have 1+ sites. This includes /plugins.

Without siteSelection :
* For users without sites: plugin listings would be incomplete (e.g. no price) and malfunctioning (the Manage Sites button is broken), as shown in the screenshot.
* For users with sites: if the load one of the /plugins/:site routes directly, Calypso could not select their site, and everything would be broken.
tl;dr not a quick fix, also affecting other screens :slightly_smiling_face:

The biggest thing that gives me pause is identifying all the little things that can break or that should change.
Say for example the “Manage sites” button: that’s an obvious thing to test, but what about other links on the page? Are the breadcrumbs working? Should we show the plugin price?
And so on.

**Comment by autumnfjeld** [comment]
Note that for this example https://www.loom.com/share/f71deadefce943628128c760a6d0d8db?sid=cd2492cb-8ae7-4c51-a2df-4f7db119b9d3 of clicking on this link from a google search:

<img width="562" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5665959/47eecbda-0a0b-4a80-a31f-be2825408361">

The desired behavior is:

* I click on the link in Google results
* I see the plugin or theme page
* I click the CTA button on a plugin listing or the CTA button on the theme listing
* I go through an onboarding steps attached to those flows

**Comment by lsl** [comment]
I don't expect it to be tackled in the same PR but if the option arises while fixing this issue, plugin screens need a 404 page https://github.com/Automattic/wp-calypso/issues/61342

**Comment by arthur791004** [comment]
Closing it as it's completed, and proposed https://github.com/Automattic/wp-calypso/pull/82658 to address why the 404 error is not displayed.
